buildSrc/subprojects/binary-compatibility/src/main/groovy/gradlebuild.binary-compatibility.gradle

```diff
@@ -84,55 +103,35 @@ dependencies {
         to.attribute(ARTIFACT_TYPE, 'gradle-classpath')
     }
-
-    // TODO update this to use a dependency on :distributionFull instead of 'rootProject.subprojects { }'
-    rootProject.subprojects { subproject ->
-        if (subproject.name == 'distributionsDependencies') {
-            currentClasspath platform(subproject)
-        }
-        plugins.withId('gradlebuild.distribution.api') {
-            apiSourceFolders.from(subproject.sourceSets.main.allJava.sourceDirectories)
-            if (subproject.plugins.hasPlugin("kotlin")) {
-                apiSourceFolders.from(subproject.sourceSets.main.kotlin.sourceDirectories)
-            }
-            newArchivesFiles.from(subproject.jar)
-            // TODO: Current classpath should be derived from the other "Gradle runtime" sources like api-metadata, runtime-api, distribution building, etc...
-            currentClasspath project(subproject.path)
-
-            // This transform uses the result of the exploded zip extraction
-            // and returns a single jar file based on the lookup project name
-            String projectName = subproject.moduleIdentity.baseName.get() - 'gradle-'
-            registerTransform(FindGradleJar) {
-                from.attribute(ARTIFACT_TYPE, 'gradle-libs-dir')
-                to.attribute(ARTIFACT_TYPE, projectName)
-                parameters {
-                    target = projectName
-                }
-            }
-            def appendixName = projectName.split('-')*.capitalize().join('')
-            def baselineConf = configurationsContainer.create("japicmp-baseline-${appendixName}") {
-                extendsFrom configurationsContainer.baseline
-                attributes.attribute(ARTIFACT_TYPE, projectName)
-            }
-            oldArchivesFiles.from(baselineConf)
-        }
-    }
 }
 
 def checkBinaryCompatibility = tasks.register("checkBinaryCompatibility", JapicmpTask) {
     def baseVersion = moduleIdentity.version.map { it.baseVersion.version }
     def isSnapshot = moduleIdentity.snapshot
-
-    inputs.files(configurations.currentClasspath)
     inputs.property('acceptedViolations', acceptedViolations.toAcceptedChangesMap())
     inputs.property("baseline.version", compatibilityBaselineVersion)
     inputs.property("currentVersion", baseVersion)
+    def apiSourceFolders = configurations.currentSources.incoming.artifactView { lenient(true) }.files
     inputs.files("apiSourceFolders", apiSourceFolders)
+    def currentClasspath = configurations.currentApiClasspath.incoming.artifactView { lenient(true) }.files
+    inputs.files(currentClasspath)
 
+    newClasspath = currentClasspath
     oldClasspath = configurations.baselineClasspath
-    newClasspath = configurations.currentClasspath
-    oldArchives = oldArchivesFiles
-    newArchives = newArchivesFiles
+
+    def currentDistributionJars = currentClasspath.filter { it.name.startsWith('gradle-') && it.name.endsWith('.jar') }
```

Comment:
I don't like this, but so far I have not been able to find a better way.

Reply:
I think this is fine.

It's an issue in other use cases as well. Maybe a limitation in Gradle if you will:
At this point, you only have the files and no other metadata anymore. So you need to filter based only on the file names (or content). So we have the "gradle-*" naming convention to identify what on the classpath belongs to Gradle and we use that here.